### PR TITLE
Upgrade to .net 6

### PR DESCRIPTION
**Pull request by taooceros**
It's pretty easy to upgrade (just replace all net5.0 with net6.0), so I don't see a reason we don't want to upgrade.

### CODE REVIEW

**Comment by jjw24** [comment]
This need to be delayed to later.

- dont upgrade Flow.Launcher.Plugin, otherwise we will need to do another force upgrade on plugins
- this will be a force upgrade for users who are not upgrading to the next release immediately, otherwise they wont be able to use the new plugins using the new api. Also means we need to repoint the manifest default branch.

personally feel a bit too soon to do so considering we just done it for 1.8.0

**Comment by jjw24** [comment]
also those that are using this pr build, please list what areas have been tested, so reviewer doesnt have to redo all.

**Comment by taooceros** [comment]
It's similar to our dotnet 5 upgrade. Plugin will work even though they don't upgrade.

**Comment by taooceros** [comment]
yeah it's true that it may be a bit early. Appveyor hasn't finished their visual studio 2022 image🤣. However, it's easy to do, so I just create one for test.

**Comment by taooceros** [comment]
> This need to be delayed to later.
> 
> * dont upgrade Flow.Launcher.Plugin, otherwise we will need to do another force upgrade on plugins
> * this will be a force upgrade for users who are not upgrading to the next release immediately, otherwise they wont be able to use the new plugins using the new api. Also means we need to repoint the manifest default branch.
> 
> personally feel a bit too soon to do so considering we just done it for 1.8.0

that probably means we take half year to finish the 1.8.0. Probably a bit slow🤣

**Comment by jjw24** [comment]
Yeah true, but to others, we just done a force, if we do this it's going to feel like another force update so soon.

**Comment by jjw24** [comment]
> It's similar to our dotnet 5 upgrade. Plugin will work even though they don't upgrade.

Yeah they will work but there is potential that 1.8.x users download new plugins with the new api then will break. Any plugins built with 6 will also break for these users.

**Comment by taooceros** [comment]
> > It's similar to our dotnet 5 upgrade. Plugin will work even though they don't upgrade.
> 
> Yeah they will work but there is potential that 1.8.x users download new plugins with the new api then will break. Any plugins built with 6 will also break for these users.

hm that's true

**Comment by taooceros** [comment]
@jjw24 I think it is possible to add multiple framework target here. So there's no issue of forcing people to use net6.0. We can remove it in the future.

**Comment by jjw24** [comment]
interesting. let's test it out

File: Flow.Launcher/Properties/PublishProfiles/Net5.0-SelfContained.pubxml [hunk 1]
```diff
@@ -1,18 +0,0 @@
...
-    <Platform>Any CPU</Platform>
-    <TargetFramework>net5.0-windows10.0.19041.0</TargetFramework>
-    <PublishDir>..\Output\Release\</PublishDir>
-    <RuntimeIdentifier>win-x64</RuntimeIdentifier>
-    <SelfContained>true</SelfContained>
-    <PublishSingleFile>False</PublishSingleFile>
-    <PublishReadyToRun>False</PublishReadyToRun>
-    <PublishTrimmed>False</PublishTrimmed>
-  </PropertyGroup>
-</Project>
```
Comment (jjw24) [inline Flow.Launcher/Properties/PublishProfiles/Net5.0-SelfContained.pubxml:18]: hey @taooceros why is this publish profile removed? we still need it right for 6?

Comment (taooceros) [reply Flow.Launcher/Properties/PublishProfiles/Net5.0-SelfContained.pubxml:18]: I remember I rename it? I will check again.

Comment (jjw24) [reply Flow.Launcher/Properties/PublishProfiles/Net5.0-SelfContained.pubxml]: all good, sorted out, see commit.

**Comment by taooceros** [comment]
.Net 7 may be released in November. Do we want to directly go to .Net 7 if we are not planning to release soon?

**Comment by jjw24** [comment]
@taooceros can you try doing a fresh build of this branch locally, are you getting any build errors?

**Comment by taooceros** [comment]
I can build it with rider, but seems visual studio 2022 may create issue. Let me try the dotnet cli.

**Comment by taooceros** [comment]
```
Fody: An unhandled exception occurred:
1>MSBUILD : error : Exception:
1>MSBUILD : error : The process cannot access the file 'D:\Document\Dotnet\Flow.Launcher\Flow.Launcher.Plugin\obj\Debug\Flow.Launcher.Plugin.dll' because it is being used by another process.
1>MSBUILD : error : Type:
1>MSBUILD : error : System.IO.IOException
1>MSBUILD : error : StackTrace:
1>MSBUILD : error :    at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
1>MSBUILD : error :    at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
1>MSBUILD : error :    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
1>MSBUILD : error :    at Mono.Cecil.ModuleDefinition.ReadModule(String fileName, ReaderParameters parameters) in D:\Code\Fody\cecil\Mono.Cecil\ModuleDefinition.cs:line 1103
1>MSBUILD : error :    at InnerWeaver.ReadModule(String assemblyFilePath, IAssemblyResolver assemblyResolver) in C:\projects\fody\FodyIsolated\ModuleReader.cs:line 30
1>MSBUILD : error :    at InnerWeaver.ReadModule() in C:\projects\fody\FodyIsolated\ModuleReader.cs:line 11
1>MSBUILD : error :    at InnerWeaver.Execute() in C:\projects\fody\FodyIsolated\InnerWeaver.cs:line 99
1>MSBUILD : error : Source:
1>MSBUILD : error : mscorlib
1>MSBUILD : error : TargetSite:
1>MSBUILD : error : Void WinIOError(Int32, System.String)
1>MSBUILD : error :
1>Done building project "Flow.Launcher.Plugin.csproj" -- FAILED.
```
Maybe just environmental error?

**Comment by taooceros** [comment]
I can build it successfully on vs2022

**Comment by jjw24** [comment]
I am getting the Cody error where the DLL is locked down by another process, tried doing a clean of both project and the .nuget folder as well as restarting VS2022. Did you get this error?

**Comment by taooceros** [comment]
> I am getting the Fody error where the DLL is locked down by another process, tried doing a clean of both project and the .nuget folder as well as restarting VS2022. Did you get this error?

I did. But it disappear after a restart.

**Comment by jjw24** [comment]
In VS2022 with the Plugin project, I can build the entire project without issues if the target is only for .Net 6 - `<TargetFrameworks>net6.0-windows</TargetFrameworks>`, as soon as I revert back to `<TargetFrameworks>net6.0-windows;net5.0-windows</TargetFrameworks>`, I get this error on every rebuild:
image

**Comment by jjw24** [comment]
I am getting the above error consistently with two separate machines. I don't think it's necessary to target .Net 5, if any user is having issues later on running older flow version installing plugins using the new .Net 6 plugin API we can just ask them to upgrade Flow.  

I will remove the .Net 5 target framework.

**Review by jjw24 — APPROVED** [review]

**Pull request merged**